src/createExecutor.ts

```diff
@@ -32,39 +31,15 @@ export function createExecutor<
 
     const body = JSON.stringify({ query, variables })
     const headers = Object.fromEntries(request.headers.entries())
-
     if (request.headers.get('accept') === 'text/event-stream') {
-      return observableToAsyncIterable({
-        subscribe: ({ next, complete, error }) => {
-          fetchEventSource(request.url, {
-            method: 'POST',
-            body,
-            headers,
-            fetch: durableObject.fetch.bind(durableObject),
-
-            onClose: () => complete(),
-            onMessage: (message) => {
-              // ping
-              if (message.data === '' && message.event === '') {
-                return
-              }
-
-              if (message.data) {
-                next(JSON.parse(message.data))
-
-                if (message.event === 'complete') {
-                  complete()
-                }
-              }
-            },
-            onError: (e) => error(e),
-          })
-
-          return {
-            unsubscribe: () => undefined,
-          }
-        },
-      })
+      // redirect stream from DO directly to client without parsing to keep stream connection client <-> DO
+      // @ts-ignore
+      context.___stream_response = await durableObject.fetch(request.url, {
+        method: 'POST',
+        body,
+        headers,
+      });
+      return {} as any;
```

Comment:
This looks intriguing, can you explain a little bit more about how this does replace the flow from before?

Reply:
@dan-lee  The existing flow is unwrapping the eventstream in the worker, and re-encoding it to send to the client. This should not be needed, the worker just needs pass the DO response to the client. It's my understanding that the current implementation will keep both the DO instance & the worker reciving the initial response active for the length of the stream. While if the response is just proxied only the DO will be kept alive.